Tutorials/Cache-Static-Assets/index.md

```diff
@@ -0,0 +1,99 @@
+---
+versionFrom: 9.3.0
+---
+
+# Cache static assets
+
+Example to cache static assets by file extension, but excluding Umbraco BackOffice assets.
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.Hosting;
+using Umbraco.Extensions;
+
+public class StaticFilesComposer : IComposer
+{
+    public void Compose(IUmbracoBuilder builder)
+        => builder.Services.AddTransient<IConfigureOptions<StaticFileOptions>, ConfigureStaticFileOptions>();
+
+    private class ConfigureStaticFileOptions : IConfigureOptions<StaticFileOptions>
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly GlobalSettings _globalSettings;
+        private static readonly HashSet<string> _fileExtensions = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            ".ico",
+            ".css",
+            ".js",
+            ".svg",
+            ".woff2"
+        };
+
+        public ConfigureStaticFileOptions(IOptions<GlobalSettings> globalSettings, IHostingEnvironment hostingEnvironment)
+        {
+            _globalSettings = globalSettings.Value;
+            _hostingEnvironment = hostingEnvironment;
+        }
+
+        public void Configure(StaticFileOptions options)
+            => options.OnPrepareResponse = ctx =>
+            {
+                string backOfficePath = _globalSettings.GetBackOfficePath(_hostingEnvironment);
+
+                // Exclude Umbraco backoffice assets
+                if (ctx.Context.Request.Path.StartsWithSegments(backOfficePath))
+                {
+                    return;
+                }
+
+                // Set headers for specific file extensions
+                var fileExtension = Path.GetExtension(ctx.File.Name);
+                if (_fileExtensions.Contains(fileExtension))
+                {
+                    var headers = ctx.Context.Response.GetTypedHeaders();
+
+                    // Update or set Cache-Control header
+                    var cacheControl = headers.CacheControl ?? new CacheControlHeaderValue();
```

Comment:
Consider using explicit type here

```suggestion
                    CacheControlHeaderValue cacheControl = headers.CacheControl ?? new CacheControlHeaderValue();
```

Reply:
@kjac in that case we could also just write:

```
CacheControlHeaderValue cacheControl = headers.CacheControl ?? new();
```